backend/src/main/java/co/inventorsoft/scripty/repository/specification/Filter.java

```diff
@@ -0,0 +1,63 @@
+package co.inventorsoft.scripty.repository.specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import org.springframework.data.jpa.domain.Specification;
+
+import co.inventorsoft.scripty.model.entity.Project;
+import co.inventorsoft.scripty.model.entity.User;
+
+
+public class Filter {
+	
+	public static Specification<Project> getFilter(User user) {
+		
+		String role = user.getRole();
+		//Long userId = user.getId();
+		
+		if (role.equals("ROLE_USER")) {
+			
+			return new Specification<Project>() {
+	            @Override
+	            public Predicate toPredicate(Root<Project> root,
+	                    CriteriaQuery<?> criteriaQuery,
+	                    CriteriaBuilder criteriaBuilder) {
+	                return criteriaBuilder.or(criteriaBuilder.equal(root.get("visibility"), 1), 
+	                		criteriaBuilder.equal(root.get("user").get("id"), user.getId()));
+	                		
+	             		
+	            }
+	        };		
+		
+		} else if (role.equals("ROLE_ADMIN")) {
+			
+			return new Specification<Project>() {
+	            @Override
+	            public Predicate toPredicate(Root<Project> root,
+	                    CriteriaQuery<?> criteriaQuery,
+	                    CriteriaBuilder criteriaBuilder) {
+	                return criteriaBuilder.greaterThanOrEqualTo(root.get("id"), 1); 	                		

```

Comment:
I don't understand how this works. If id is generated by the database after project insert we will have all project ids greater than or equal to 1. This means the following predicate is always evaluates to true. Why do we need it then?